### PR TITLE
fix(ILP): Consume bytes left in a buffer after a successful authentication

### PR DESCRIPTION
**Pull request by jerrinot**
When authentication is completed and there are still bytes remaining in the buffer
we have to consume them. Otherwise, they might be stuck there until there is
a new READ EVENT. 

Ideas on how to test this are welcome!

### CODE REVIEW

**Comment by puzpuzpuz** [comment]
> Ideas on how to test this are welcome!

Do same thing as Go client tests did before this commit: https://github.com/questdb/go-questdb-client/commit/2c81786c009844aa7dbeed77da66ec6eb9c06f15 ?

File: core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContext.java [hunk 1]
```diff
@@ -86,7 +86,14 @@ IOContextResult handleIO(NetworkIOJob netIoJob) {
         if (authenticated) {
             return super.handleIO(netIoJob);
         }
-        return handleAuth();
+        IOContextResult result = handleAuth();
+        if (authenticated && recvBufPos > recvBufStart) {
+            // authentication is completed and there are still bytes remaining in the buffer
```
Comment (ideoma) [inline core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContext.java:91]: I think `recvBufPos > recvBufStart` does not mean bytes remain in the buffer. I don't see `recvBufPos` set in `findLineEnd()`.
I think this condition will be true always after signed challenge is received.

Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContext.java:91]: > I don't see `recvBufPos` set in `findLineEnd()`.

It's set in `waitForResponse()` -> `compactBuffer()`

**Comment by bluestreak01** [comment]
@jerrinot to test, you can create a loop, which will be sending challenge response followed by complete ILP line

**Comment by bluestreak01** [comment]
btw, the fix is incorrect, a puzzler :)

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContext.java|3|3|100.00%|

**Comment by bluestreak01** [comment]
@glasstiger  committed a spoiler :) Basically we need to call `parseMeasurement` if there is non-empty buffer. Calling `handleIO` will work when ILP is incomplete and might hang with line is complete.

File: core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContextTest.java [hunk 2]
```diff
@@ -168,6 +168,28 @@ public void testGoodAuthentication() throws Exception {
...
+    @Test
+    public void testGoodAuthenticationWithExtraData() throws Exception {
+        runInAuthContext(() -> {
+            try {
+                boolean authSequenceCompleted = authenticate(AUTH_KEY_ID1, AUTH_PRIVATE_KEY1,
+                        "weather,location=us-midwest temperature=82 1465839830100400200\n");
+                Assert.assertTrue(authSequenceCompleted);
+            } catch (RuntimeException ex) {
+                // Expected that Java 8 does not have SHA256withECDSAinP1363
+                if (ex.getCause() instanceof NoSuchAlgorithmException && TestUtils.getJavaVersion() <= 8) {
```
Comment (jerrinot) [inline core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContextTest.java:180]: @glasstiger: on what Java you were getting this error? Was it in the CI? 
I tried it on Azul Zulu 8u332 and it does have the algorithm.

Comment (jerrinot) [inline core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContextTest.java:180]: @glasstiger: on what Java you were getting this error? Was it in the CI? 
I tried it on Azul Zulu 8u332 and it does have the algorithm.

Comment (glasstiger) [reply core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContextTest.java:180]: that is a copy-paste from testGoodAuthentication().
i copied the test and modified it to send the ILP line together with the auth message.

Comment (glasstiger) [reply core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContextTest.java:180]: looks like @ideoma added this try-catch 2 months ago

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContext.java|3|3|100.00%|

File: core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContextTest.java (hunk 2 above)
Comment (bluestreak01) [reply core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContextTest.java:180]: tagging @ideoma

Comment (ideoma) [reply core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpAuthConnectionContextTest.java:180]: It didn't work on some Java 8 version a year ago. I don't remember which version exactly.
Free to remove it if you can run the test on latest Java 8.

**Review by ideoma — APPROVED** [review]

**Pull request merged**